src/macaron/dependency_analyzer/repo_finder.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module tries to find urls of repositories that match artifacts passed in 'group:artifact:version' form."""
+
+import logging
+import re
+import typing
+
+import requests
+from defusedxml.ElementTree import fromstring
+
+from macaron.config.global_config import global_config
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+def create_urls(group: str, artifact: str, version: str) -> list[str]:
+    """
+    Create the urls to search for the pom relating to the passed GAV.
+
+    Parameters
+    ----------
+    group : str
+        The group ID.
+    artifact: str
+        The artifact ID.
+    version: str
+        The version of the artifact.
+
+    Returns
+    -------
+    list[str]
+        The list of created URLs.
+    """
+    urls = []
+
+    version_pruned = ""
+    if "-" in version:
+        # Create a pruned version to fix some strangely labeled artifacts
+        version_pruned = re.sub("-[a-zA-Z]+", "", version)
+
+    # Create URLs using all configured repositories
+    for repo in global_config.artifact_repositories:
+        urls.append(f"{repo}/{group}/{artifact}/{version}/{artifact}-{version}.pom")
```

Comment:
Yes. The detection method for strange versions isn't robust, and really only exists because I found one case where it was needed to succeed. I would say there is a still a chance the non pruned version is correct when the pruning is triggered.

Reply:
Do you remember the case where this was needed?

Reply:
I checked again and it seems none of the versions that were pruned led to a successful POM retrieval. I'll remove this.